### PR TITLE
Make `NavigationView` publicly usable.

### PR DESCRIPTION
**Pull request by MaximAlien**
### Description

This PR makes modifications that are required to make `NavigationView` publicly usable as a standalone view.

New `NavigationViewExample` was also added for testing purposes.

Example that showcases such usage can be found in: https://github.com/mapbox/mapbox-navigation-ios-examples/pull/172.

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationView.swift [hunk 1]
```diff
@@ -172,9 +172,9 @@ open class NavigationView: UIView {
     
     lazy var speedLimitView: SpeedLimitView = .forAutoLayout(hidden: true)
     
-    lazy var topBannerContainerView: BannerContainerView = .forAutoLayout()
+    public lazy var topBannerContainerView: BannerContainerView = .forAutoLayout()
```
Comment (Udumft) [inline Sources/MapboxNavigation/NavigationView.swift:175]: Is it safe to provide readwrite access to these properties?

File: Sources/MapboxNavigation/NavigationView.swift [hunk 2]
```diff
@@ -254,6 +254,7 @@ open class NavigationView: UIView {
     }
     
     func commonInit() {
+        DayStyle().apply()
```
Comment (bamx23) [inline Sources/MapboxNavigation/NavigationView.swift:257]: Is it required?

**Comment by MaximAlien** [comment]
@Udumft, @bamx23, thanks for initial look. This PR is still in draft and more modifications will be required (I've created it for https://github.com/mapbox/mapbox-navigation-ios-examples/pull/172), I'll let you know when it's ready for review.

File: Sources/MapboxNavigation/NavigationView.swift (hunk 2 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationView.swift:257]: This is the easiest way to apply styling for the views in `NavigationView`, as some of them are blank. Users are able to change colors to anything else as well.

File: Sources/MapboxNavigation/NavigationView.swift (hunk 1 above)
Comment (MaximAlien) [reply Sources/MapboxNavigation/NavigationView.swift:175]: Idea is to allow users use `NavigationView` independently. This change will allow them to modify top and bottom banners.

**Comment by MaximAlien** [comment]
Merging this PR into `maxim/1378-add-preview-view-controller` to continue work there.

**Pull request merged**